docs/content/concepts/metadata-tags/asset-metadata/table-metadata.mdx

```diff
@@ -0,0 +1,192 @@
+---
+title: "Table metadata | Dagster Docs"
+description: "Table metadata can be used to provide additional context about a tabular asset, such as its schema, row count, and more."
+---
+
+# Table metadata
+
+Table metadata provides additional context about a tabular asset, such as its schema, row count, and more. This metadata can be used to improve collaboration, debugging, and data quality in your data platform.
+
+Dagster supports attaching different types of table metadata to assets, including:
+
+- [**Column schema**](#attaching-column-schema): Describes the structure of the table, including column names and types
+- [**Row count**](#attaching-row-count): Describes the number of rows in a materialized table
+- [**Column-level lineage**](#attaching-column-level-lineage): Describes how a column is created and used by other assets
+
+---
+
+## Attaching column schema
+
+### For assets defined in Dagster
+
+<Image
+alt="Column schema for an asset in the Dagster UI"
+src="/images/concepts/metadata-tags/metadata-table-schema.png"
+width={1793}
+height={652}
+/>
+
+You may attach column schema metadata to Dagster assets either as [definition metadata](/concepts/metadata-tags/asset-metadata#attaching-definition-metadata) or [materialization metadata](/concepts/metadata-tags/asset-metadata#attaching-materialization-metadata). If the schema of your asset is pre-defined, you can attach it as definition metadata. If the schema is only known when an asset is materialized, you can attach it as metadata to that materialization.
+
+To attach schema metadata to an asset, you will need to:
+
+1. Construct a <PyObject object="TableSchema"/> object with <PyObject object="TableColumn"  /> entries describing each column in the table
+2. Attach the `TableSchema` object to the asset as part of the `metadata` parameter under the `dagster/column_schema` key. This can be attached to your asset definition, or to the <PyObject object="MaterializeResult" /> object returned by the asset function
+
+Below are two examples of how to attach column schema metadata to an asset, one as definition metadata and one as materialization metadata:
```

Comment:
Given that we're using the terms **definition metadata** and **materialization metadata**, could we work these into the code comments to make it easy to skim and still tell which is which?

Ex:

```
# Definition metadata
# Here, we know the schema of the asset...
```